cartography/intel/github/repos.py

```diff
@@ -214,6 +226,58 @@ def _transform_collaborators(collaborators, repo_url, transformed_collaborators)
             transformed_collaborators[user_permission].append(user)
 
 
+def _transform_python_requirements(repo_object, out_requirements_files):
+    """
+    Performs data transformations for the requirements.txt files in a GitHub repo, if available.
+    :param repo_object: The repo object.
+    :param out_requirements_files: Output array to append transformed results to.
+    :return: Nothing.
+    """
+    req_file_contents = repo_object['requirements']
+    if req_file_contents and req_file_contents.get('text'):
+        # Remove --no-binary tags if present
+        text_contents = req_file_contents['text']
+        text_contents.replace('--no-binary ', '')
+
+        parsed_list = []
+        for line in text_contents.split("\n"):
+            try:
+                # Skip over empty lines
+                if line:
+                    req = Requirement(line)
+                    parsed_list.append(req)
+            except ValueError as e:
+                logger.info(
+                    f"Failed to parse line \"{line}\" in repo {repo_object['url']}'s requirements.txt; skipping. "
+                    f"Details: {e}.",
+                )
+                continue
+
+        for req in parsed_list:
+            if not req.name:
+                logger.info(
+                    f"Could not get library name for an object in {repo_object['url']}'s requirements.txt, skipping.",
+                )
+                continue
+
+            # Set `spec` to a default value. Example values for str(req.specifier): "<4.0,>=3.0" or "==1.0.0".
+            spec = str(req.specifier)
+
+            if len(req.specifier._specs) == 1:
+                # req.specifier._specs is a frozenset so to manipulate it we turn it into a list and get the 1st item.
+                spec_str = str(list(req.specifier._specs)[0])
+                # If the spec is pinned to 1 version with ==, we only want the version number, not the full specifier.
+                spec = spec_str.strip('==') if spec_str.startswith('==') else spec
+
+            out_requirements_files.append({
+                "id": f"{req.name}|{spec}",
+                "name": req.name,
```

Comment:
We will want to canonicalize the package name for the node; packaging has a function we can leverage here: https://packaging.pypa.io/en/latest/utils/#packaging.utils.canonicalize_name.
This ensures multiple repos depending on the same version of a library map to the same node even if they spell (e.g. capitalize) it differently.

Per our chat we discussed potentially putting the specifier as an attribute on the edge; could be reasonable to leave the original naming (and e.g. markers) there.